### DISCUSSION TITLE
flink cdc 如何保证binlog采集的有序性

### DISCUSSION THREAD

**Discussion by sanguoasd** [apache/flink-cdc#1331] [Q&A]
flink cdc在采集binlog的增量数据时，源端binlog可能是有序的，但是在后续传输过程中，会出现乱序的现象吗？以及如果出现了如何处理这种情况？

**Comment by voyagertanyao** [comment]
binlog增量阶段只支持单并发模式，所以cdc不存在时序性问题的。
像我们业务中，采用Datastream API的方式采集binlog，然后写入kafka，为了保证数据的有序性，我们额外解析了binlog中三个字段：
1. 表主键；
2. binlog发生时间（ts_ms）
3. binlog消息位点（pos）
通过这三个参数的比较来保证数据的有序性的
主键相同的必须放在同一个kafka topic中
